sections/featured-product.liquid

```diff
@@ -83,7 +95,9 @@
                     loop: section.settings.enable_video_looping,
                     modal_id: section.id,
                     xr_button: false,
-                    media_width: 0.5
+                    media_width: media_width,
```

Comment:
I believe the thought was that would be too "magic". Especially considering we offer the media width setting to manually control that. The other risk would be the potentially large amount of layout shift when changing variants if the featured images varied in dimension at all.

Reply:
Yeah we would have to rework a lot of things to make it work that way. I feel like both ways are "magical" though. Ill be curious to see how it's used and received by merchants 🤔 
I think some great descriptive explanation of what `Constrain media to screen height` means and how it might impact things will be useful on our help docs.

Reply:
I did like things about that particular solution (if we're talking about the same thing), but it was ruled out for reasons beyond refactor. I'm not sure we can refactor our way out of the layout shift issue without making other weird assumptions about what determines column width in certain scenarios. Though obviously the line to draw between what is _too_ magic is subjective and ultimately on us to decide and validate with merchant frustration.